### ISSUE TITLE
Constricting Ring causes floating damage numbers after death

### ISSUE THREAD

**Issue by kphoenix137** [diasurgical/devilutionX#6274]
### Operating System

Windows x64

### DevilutionX version

1.5.0 (latest release)

### Describe

title

### To Reproduce

put constricting ring on some appendage

die

https://github.com/diasurgical/devilutionX/assets/68359262/0901e052-7cd1-4d71-8a31-6c1e252ee106

### Expected Behavior

no floating numbers

**Comment by qndel** [comment]
Cannot reproduce

**Comment by StephenCWills** [comment]
> Cannot reproduce

It seems that Constricting Ring counts as a monster kill so your gear will drop to the floor if you let it kill you. Try dying to a Firewall or just go to dlvl 16 to prevent gear from dropping.

**Comment by kphoenix137** [comment]
It's likely I mislabeled the version here. I think this occurred when I was playtesting custom death messages. Adding these death messages and showing the source of the death required me to refactor `DeathReason`, in which I made Constricting Ring death no longer be label as a monster death. That's how I didn't drop the ring. However like staphen said, if you die to a non-monster source while equipped with this ring, then it would still show floating numbers after death. I'll look into this.
